File: Sorting_algorithms/Python/Selection_sort.py
# Time:
# Best: O(n ^ 2)
# Average: O(n ^ 2)
# Worst: O(n ^ 2)

# Space: O(1)
def selectionSort(arr, n):
    
    for i in range(n):
        minIndex = i
        
        for j in range(i + 1, n):
            if arr[minIndex] > arr[j]:
                minIndex = j
        
        if minIndex != i:
            arr[minIndex], arr[i] = arr[i], arr[minIndex]

File: Sorting_algorithms/C/Selection_sort.c
// Time:
// Best: O(n ^ 2)
// Average: O(n ^ 2)
// Worst: O(n ^ 2)

// Space: O(1)
void selectionSort(int* arr, int n) {
    
    for(int i = 0; i < n; i++) {
        int minIndex = i;
        
        for(int j = i + 1; j < n; j++) {
            if(arr[j] < arr[minIndex]){
                minIndex = j;
            }
        }
        
        if(minIndex != i) {
            arr[minIndex] += arr[i];
            arr[i] = arr[minIndex] - arr[i];
            arr[minIndex] -= arr[i];
        }
    }
}
